egs/librispeech/ASR/pruned_transducer_stateless/decoder.py

```diff
@@ -50,13 +51,25 @@ def __init__(
           context_size:
             Number of previous words to use to predict the next word.
             1 means bigram; 2 means trigram. n means (n+1)-gram.
+          modify_embedding_layer:
+            Do or not do modify the decoder embedding layer.
         """
         super().__init__()
         self.embedding = nn.Embedding(
             num_embeddings=vocab_size,
             embedding_dim=embedding_dim,
             padding_idx=blank_id,
         )
+        self.modify_embedding_layer = modify_embedding_layer
+        if self.modify_embedding_layer:
+            self.embedding.weight = nn.Parameter(
+                (
+                    torch.randn(vocab_size, embedding_dim)
+                    * (embedding_dim ** -0.5)
+                ).detach()
+            )
+
+        self.embedding_dim = embedding_dim
```

Comment:
```suggestion
        self.embedding_dim_inv_sqrt = embedding_dim ** -0.5
        embedding_weight = torch.randn(vocab_size, embedding_dim) * self.embedding_dim_inv_sqrt
        self.embedding = nn.Embedding(
            num_embeddings=vocab_size,
            embedding_dim=embedding_dim,
            padding_idx=blank_id,
            _weight=embedding_weight
        )
        self.embedding._fill_padding_idx_with_zero()
```

And change line 99 to
```python3
embedding_out = self.embedding(y) * self.embedding_dim_inv_sqrt
```

Reply:
I think is self.embedding(y) * (embedding**0.5). The self.embedding_dim_inv_sqrt is embedding**-0.5

Reply:
Oh, yes, you're right. But please make it a member of `self` so that it is computed only once.